server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/Node.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.core.model.tree;
+
+import org.eclipse.lsp4j.Location;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * The class represents a Node in source structure tree.
+ */
+public abstract class Node {
+  private final Location location;
+  private Node parent;
+  private final List<Node> children = new ArrayList<>();
+  private final NodeType nodeType;
+
+  protected Node(Location location, NodeType nodeType) {
+    this.location = location;
+    this.nodeType = nodeType;
+  }
+
+  public Location getLocation() {
+    return location;
+  }
+
+  public Node getParent() {
+    return parent;
+  }
+
+  public List<Node> getChildren() {
+    return children;
+  }
+
+  /**
+   * Adds children node to this node and updates children parent link.
```

Comment:
I see some mess with the method javadocs here and in other classes, it should start with a verb in the active voice,